pkg/oc/admin/policy/modify_roles.go

```diff
@@ -320,6 +322,17 @@ func (o *RoleModificationOptions) CompleteUserWithSA(f *clientcmd.Factory, cmd *
 		return errors.New("you must specify at least one user or service account")
 	}
 
+	// return an error if a fully-qualified service-account name is used
+	for _, sa := range saNames {
+		if strings.HasPrefix(sa, "system:serviceaccount") {
+			return errors.New("--serviceaccount (-z) should only be used with short-form serviceaccount names (e.g. \"default\")")
+		}
+
+		if len(validation.ValidateServiceAccountName(sa, false)) > 0 {
```

Comment:
Why suppress the actual errors?

Reply:
Saw this done in at least one other place, so copied it; will store and append the errors to the final message

Reply:
Updated message to show indented causes in new lines:

```
$ oc policy add-role-to-user admin -z :default
error: ":default" is not a valid serviceaccount name:
  a DNS-1123 subdomain must consist of lower case alphanumeric characters, '-' or '.', and must start and end with an alphanumeric character (e.g. 'example.com', regex used for validation is '[a-z0-9]([-a-z0-9]*[a-z0-9])?(\.[a-z0-9]([-a-z0-9]*[a-z0-9])?)*')
See 'oc policy add-role-to-user -h' for help and examples.
```